site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -146,69 +184,251 @@ def _transform_publication_date(self):
+        #     "proceedings": "publication-conferenceproceeding",
+        #     "report": "publication-report",
+        #     "thesis": "publication-thesis",
+        # }
+        #
+        # if document_type not in document_type_mapping:
+        #     self.metadata_errors.append(
+        #         f"Error occurred while mapping document_type to resource_type. Couldn't fine a mapping rule for "
+        #         f"document_type {document_type}. INSPIRE record id: {self.inspire_metadata.get('control_number')}."
+        #     )
+        #     return None
+
+        # return document_type_mapping.get(document_type)
+
+        document_types = self.inspire_metadata.get("document_type", [])
+        for doc_type in document_types:
+            if doc_type == "article":
+                self.metadata_errors.append("Articles are not supported for now.")
+                return None
+
+        # uncomment the part above for other doc types (thesis have only 1 possible doc type)
+        return "publication-thesis"
+
+    def _transform_contributors(self):
+        """Mapping of INSPIRE authors to contributors."""
+        authors = self.inspire_metadata.get("authors", [])
+        contributors = []
+
+        corporate_authors = self.inspire_metadata.get("corporate_author", [])
+        mapped_corporate_authors = []
+        for corporate_author in corporate_authors:
+            contributor = {
+                "person_or_org": {
+                    "type": "organizational",
+                    "name": corporate_author,
+                },
+            }
+            mapped_corporate_authors.append(contributor)
 
-        if document_type not in document_type_mapping:
-            self.metadata_errors.append(
-                f"Error occurred while mapping document_type to resource_type. Couldn't fine a mapping "
-                f"rule for document_type {document_type}. INS"
-                f"PIRE record id: {self.inspire_metadata.get('control_number')}."
-            )
-            return None
+        for author in authors:
+            inspire_roles = author.get("inspire_roles", [])
+            if "supervisor" in inspire_roles:
+                contributors.append(author)
 
-        result_doc_type = document_type_mapping.get(document_type)
-        return result_doc_type
+        return self._transform_creatibutors(contributors) + mapped_corporate_authors
 
     def _transform_creators(self):
-        """Mapping of INSPIRE authors to creators and contributors."""
+        """Mapping of INSPIRE authors to creators."""
+        authors = self.inspire_metadata.get("authors", [])
         creators = []
-        authors = self.inspire_metadata.get("authors")
+        for author in authors:
+            inspire_roles = author.get("inspire_roles")
+            if not inspire_roles:
+                creators.append(author)
+            elif "author" in inspire_roles or "editor" in inspire_roles:
```

Comment:
are we sure about editor being considered as creator of a record?

Reply:
Yes, here is a formula from Micha (also confirmed with the team):
- no role -> creator
- author -> creator, role = author
- editor -> creator, role = editor
- supervisor -> contributor, role = supervisor